### PR TITLE
Update crate

### LINKED ISSUE #34: Feature "UI_Xaml" doesn't compile with windows-rs 0.29-git
## Problem
There seems to be a change in Vtable struct names. 

Note: need to use windows-rs git to pass the missing Debug impl error

### LINKED ISSUE #35: Unidiomatic use of `Result::and_then` in README
Currently the code in the README:
```rust
fn main() -> ::windows::core::Result<()> {
    bootstrap::initialize()
        .and_then(|_| {
            println!(
                "Remaining charge: {}%",
                PowerManager::RemainingChargePercent()?
            );
            Ok(())
        })
        .and_then(|_| bootstrap::uninitialize())
}
```

I strongly suggest replacing it with the following:
```rust
fn main() -> ::windows::core::Result<()> {
    bootstrap::initialize()?;
    println!(
        "Remaining charge: {}%",
        PowerManager::RemainingChargePercent()?,
    );
    bootstrap::uninitialize()
}
```

I personally cannot think of any reason preventing the use of the `?` operator, because it is already used in the original example. I would go even further to suggest the following:
```rust
fn main() -> ::windows::core::Result<()> {
    bootstrap::initialize()?;
    let power = PowerManager::RemainingChargePercent()?;
    println!("Remaining charge: {power}%");
    bootstrap::uninitialize()
}
```

### PR DESCRIPTION
**Pull request by riverar**
Adds compatibility for `windows` (0.34), samples, and interop helpers to unwrap Microsoft.UI types for GDI compatibility.

Fixes #34, fixes #35

### CODE REVIEW

**Comment by kennykerr** [comment]
Cargo.lock should be gitignore'd for library crates.

File: Cargo.toml [hunk 1]
```diff
@@ -51,233 +53,74 @@ Graphics_DirectX = [
...
   "UI_Composition",
-  "windows/Foundation",
-  "windows/Foundation_Collections",
-  "windows/Foundation_Numerics",
-  "windows/Graphics",
-  "windows/Graphics_Effects",
-  "windows/UI",
 ]
 UI_Composition_Effects = [
   "UI_Composition",
```
Comment (kennykerr) [inline Cargo.toml:68]: Consider flattening these to a single line as the windows crate does.

**Comment by kennykerr** [comment]
> pub mod interop;

Not sure we need a nested module for this. Can we just put it all in the windows_app::core module? 

---
Refers to: src/core/mod.rs:1 in fb04deb. [](commit_id = fb04debe86fcd9f024f405009370c1e85b932453, deletion_comment = False)

**Comment by kennykerr** [comment]
>         HWND(window_id.Value.try_into().unwrap())

You really should be calling Windowing_GetWindowFromWindowId. :sunglasses: 

---
Refers to: src/core/interop/windowing/mod.rs:20 in fb04deb. [](commit_id = fb04debe86fcd9f024f405009370c1e85b932453, deletion_comment = False)

**Review by kennykerr — DISMISSED** [review]

**Comment by riverar** [comment]
Going to pause ⏸️ merging this until we can chat next week about some UX concerns.

File: .github/readme.md [hunk 2]
```diff
@@ -47,15 +44,10 @@ use ::windows_app::Microsoft::Windows::System::Power::*;
...
-    bootstrap::initialize()
-        .and_then(|_| {
-            println!(
-                "Remaining charge: {}%",
-                PowerManager::RemainingChargePercent()?
-            );
-            Ok(())
-        })
-        .and_then(|_| bootstrap::uninitialize())
+    bootstrap::initialize()?;
```
Comment (kennykerr) [inline .github/readme.md:47]: I take it this sample is out of date?

File: crates/samples/appwindow/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+use windows::Win32::UI::WindowsAndMessaging::{
+    DispatchMessageW, GetMessageW, GetSystemMetrics, GetWindowRect, PostQuitMessage, SetWindowPos,
+    TranslateMessage, MSG, SM_CXSCREEN, SM_CYSCREEN, SWP_NOMOVE, SWP_NOSIZE,
+};
+
+fn main() -> Result<()> {
+    bootstrap::initialize()?;
+    sample_main()?;
+    bootstrap::uninitialize()
+}
```
Comment (kennykerr) [inline crates/samples/appwindow/src/main.rs:16]:
```suggestion
fn main() -> Result<()> {
    sample_main()?;
}
```

Why not this?

File: .github/readme.md (hunk 2 above)
Comment (riverar) [reply .github/readme.md:47]: It's current. If you're thinking about winappsdk xcopy deployment, that hasn't shipped yet. 😄

File: crates/samples/appwindow/src/main.rs (hunk 3 above)
Comment (riverar) [reply crates/samples/appwindow/src/main.rs:16]: As soon as WinAppSdk 1.1 ships, definitely!

**Review by kennykerr — APPROVED** [review]

**Pull request merged**